### ISSUE TITLE
1.4.10: If original page must be scrolled and unchanged at 320px - is that a violation of "Reflow"?

### ISSUE THREAD

**Issue by JAWS-test** [w3c/wcag#2063]
At 1280 px browser width, a page has two columns of text, each column is 1280 px wide, which means you have to scroll horizontally to read the second column after the first.
At 320 px browser width, the two columns are still next to each other and are now 320 px wide. So the text can be read without scrolling, but to get to the second column you have again to scroll horizontally.
I.e. there is no difference in the effort for scrolling between the 100% view and the 400% zoom view. 
Is this still a violation of 1.4.10. According to the wording yes. But does it make sense?

**Comment by alastc** [comment]
I would say that fails reflow.

Does it make sense? Not really, but nor does the initial scenario of requiring horizontal scrolling.

**Comment by JAWS-test** [comment]
> but nor does the initial scenario of requiring horizontal scrolling.

This is a design decision that some sites use. These pages then do not have to be scrolled vertically. But this are pages for horizontally written languages.

**Comment by patrickhlauke** [comment]
so do they require vertical and horizontal scrolling once at 320 CSS px width ? if not, don't think that's an issue?

**Comment by alastc** [comment]
I thought it was where there are two columns, each set to 100% wide, so it intentionally shows one column at a time, and you have to scroll to see the second. 
They would have vertical & horizontal scrolling at all screen sizes, unless you have a giant monitor and short content.

> These pages then do not have to be scrolled vertically. 

Ah, ok, that's different then. The SC doesn't differentiate on the language direction, but the intended scrolling direction.

If you scroll horizontally (at a height equivalent to 256 CSS pixels) but not vertically, that's fine.

**Comment by JAWS-test** [comment]
> The SC doesn't differentiate on the language direction, but the intended scrolling direction.

The understanding says the opposite:

> The success Criterion applies to both horizontally and verticaly written languages. Zooming the page for horizontally written languages where pages scroll vertically by default (e.g. English) should not require horizontal scrolling. Zooming the page for vertically written languages which scroll horizontally by default should not require vertical scrolling

Some of the pages I need to test must be scrolled initially, i.e. without zooming even in both directions. This may not be nice, but that's the way it is. Even then, my question would be: if the scrolling effort in at least one direction does not increase at 400% and text blocks can be read without horizontal scrolling: Why should that be a problem?

**Comment by mraccess77** [comment]
It's sort of like a table cell in that sense as long as you can read a block of text without scrolling in both directions it could pass - but it comes down to interpretation of "Content can be presented".  Practically speaking it doesn't meet the spirit - but if you have to scroll to bring the block into view but then it fits horizontally and you have to scroll - that block can be read with scrolling in one direction - although access to the block requires scrolling in another direction.  This is pretty much the same point of the discussion about regions of content in software apps.   If you have a panel on the right that you have to scroll to reach but once you scroll to it - it all fits in 320css pixels horizontally then you don't have to scroll horizontally to read it.

What does "Content can be presented" mean? Does it mean "read" or mean "accessed"?

**Comment by alastc** [comment]
> The understanding says the opposite

The 'common' use-case for horizontally scrolling content is vertically-oriented languages, which is why the understanding document explains it that way. Above, I meant that the normative text doesn't use language as the differentiator, but it does use the intended scrolling direction.

> my question would be: if the scrolling effort in at least one direction does not increase at 400% and text blocks can be read without horizontal scrolling: Why should that be a problem?

This is essentially a variation on the reasoning that other people have complained about when there is 2-way scrolling at lower zoom levels, but until it hits the exact 320px wide it isn't a fail.

The answer is that it *is* a problem across the levels-of-zoom / widths-of-windows, but we selected a particular size to focus on with the SC. The logic being that most sites in most situations will make it work at various levels anyway, rather than just passing the letter of the SC. (And it's not a huge burden to test.)

**Comment by JAWS-test** [comment]
@alastc 

> That means it scopes-in things pages like the Netflix interface with lots of horizontally scrolling areas

That is interesting to read. I suspect that many understand 1.4.10 as requiring an entire page to wrap at 320 px so that it doesn't have to scroll horizontally. Your comment, however, now sounds to me like it is not forbidden for a page that is vertically scrolled throughout to contain areas that can be scrolled horizontally (as long as the area does not need to be scrolled vertically to perceive its content). 
If this is correct, I suggest the following:
1: Explain the case in Understanding (in the chapter "Avoiding scrolling in horizontally and vertically written languages" - the chapter then needs a new heading).
2: Describe in the Understanding that the SC always refers to the combination of 320x256 px, because the current wording sounds like that for English pages e.g. everything has to work only with 320px width and the height doesn't matter, which leads to the fact that various errors can't be evaluated (e.g. fixed header that is too big, see https://github.com/w3c/wcag/issues/2043).

**Comment by JAWS-test** [comment]
@mraccess77 

> What does "Content can be presented" mean? Does it mean "read" or mean "accessed"?

I think the question of what content is is more important. Is content a page area that contains a paragraph of text (as defined in SC 1.4.8: "blocks of text")? If so, my initial example would not be a problem at all, because I can read each individual block of text without scrolling horizontally, but only have to scroll horizontally to get from one column to the next, i.e. from one content to the next content. 
Is this what is meant by SC 1.4.10? Then why is it not formulated as clearly as in 1.4.8: "blocks of text"?

As already formulated in the previous post: I think many understand 1.4.10 to mean that a page must wrap completely at 320px (and not just individual page areas to be able to read the content without scrolling, even if that was the original intention).
